src/app/controllers/sap_portal.php

```diff
@@ -27,6 +29,9 @@ public function mission($id = null, $action = '') {
         if (isset($id) && ctype_digit($id)) {
             $mission = $this->sap_model->get_mission($id);
             if ($mission) {
+                if (!$mission['published'] && !$this->sap_auth->is_current_user_admin()) {
+                    $this->http_lib->response_code(403);
```

Comment:
We can give 404 here. Since mission IDs are continuous, someone non-admin could see if next mission is coming or not with this! :P